dialogue-core/src/main/java/com/palantir/dialogue/core/TimingEndpointChannel.java

```diff
@@ -48,10 +48,12 @@
         this.responseTimer = metrics.response()
                 .channelName(channelName)
                 .serviceName(endpoint.serviceName())
+                .endpoint(endpoint.endpointName())
                 .build();
         this.ioExceptionMeter = metrics.responseError()
                 .channelName(channelName)
                 .serviceName(endpoint.serviceName())
+                .endpoint(endpoint.endpointName())
                 .reason("IOException")
                 .build();
```

Comment:
I think we should report frequency of failures in some way without ignoring non-io exceptions. On a per-endpoint basis I'd like to know the ratio of client-perceived failures regardless(ish[1]) of the failure type.

1. We'd need to exclude cancelation exceptions since it's a failure type that the client requests.

Reply:
Do you want to add more exceptions in there as reason, or just a new counter maybe that counts everything together, except for the cancellation exceptions?

Reply:
basically, I don't have context on what is an exhaustive list of exceptions if we want to list them; could just look at the exception type, but then I'd need a cache for metric names keyed by that, which feels overkill; why do you think knowing per-reason type is useful?

Reply:
I don't have a strong preference about the reason, I think we want to get a sense for the calls that failed unexpectedly. ServiceException/RemoteException are a little bit murky, I'm not sure how we want to measure those because it's possible infra changes can cause them, but in most cases they're the result of misuse. Happy to defer any changes around those to another PR :-)

Reply:
Alright, let's merge as is and chat when you're online.